subsys/net/lib/dns/mdns_responder.c

```diff
@@ -423,7 +428,21 @@ static void send_sd_response(struct net_context *ctx,
 		service_type_enum = true;
 	}
 
-	DNS_SD_FOREACH(record) {
+	DNS_SD_COUNT(&rec_num);
+
+	while (rec_num > 0 || ext_rec_num > 0) {
+		/*
+		 * The loop will always iterate over all entries, it can be done
+		 * backwards for simplicity
+		 */
+		if (rec_num > 0) {
+			DNS_SD_GET(rec_num - 1, &record);
+			rec_num -= 1;
```

Comment:
That is perfectly valid code but usually -1 is done via `rec_num--;`

Reply:
Done